templates/hooks/sdk_addon.go.tpl

```diff
@@ -0,0 +1,24 @@
+{{ $outputShape := .CRD.GetOutputShape .CRD.Ops.Create }}
+// This method copies the data from given {{ $outputShape.ShapeName }} by populating it
+// into copy of supplied resource and returns that.
+func (rm *resourceManager) set{{ $outputShape.ShapeName }}Output (
+    r *resource,
+    obj *svcsdk.{{ $outputShape.ShapeName }},
+) (*resource, error) {
+	if obj == nil ||
+		r == nil ||
+		r.ko == nil {
+		return nil, nil
+	}
+	resp := &svcsdk.{{ .CRD.Ops.Create.OutputRef.Shape.ShapeName }}{ {{ $outputShape.ShapeName }}:obj }
+	// Merge in the information we read from the API call above to the copy of
+    // the original Kubernetes object we passed to the function
+    ko := r.ko.DeepCopy()
+{{ $createCode := GoCodeSetCreateOutput .CRD "resp" "ko" 1 true }}
+{{ $createCode }}
+	rm.setStatusDefaults(ko)
+{{- if $hookCode := Hook .CRD "sdk_addon_set_output_post_populate" }}
```

Comment:
Nit: you can also add this hook to list of known hooks (comment) in https://github.com/aws-controllers-k8s/code-generator/pull/94 - https://github.com/aws-controllers-k8s/code-generator/blob/85436ab1f0b7994e7c6651a03177ab90a8b61ad5/pkg/generate/ack/hook.go#L60

Reply:
As this dynamic template and the hook id that it consumes, is scoped to the service controller and is specified in generator config for elasticache controller.
This way any dynamic template can open its extension points by supporting such hooks.

As this hook is specific to this service controller scoped template I considered it local.